contracts/Staking.sol

```diff
@@ -290,12 +295,24 @@ contract Staking is Ownable, ReentrancyGuard {
     ) internal {
         for (uint256 index = 0; index < allowedTokens.length; index++) {
             address tokenAddress = allowedTokens[index];
-            uint256 usersStake = usersPool[tokenAddress];
-            uint256 reward = usersStake.mul(_rewardPercentage).div(_precision);
+
+            uint256 reward = providersPool[tokenAddress]
+                .div(leverage)
+                .mul(_rewardPercentage)
+                .div(_precision);
+
+            // Reward must be less than 25% of usersPool to ensure payout at all time
+            if (reward > usersPool[tokenAddress].mul(25).div(100)) {
+                reward = usersPool[tokenAddress].mul(_rewardPercentage).div(
```

Comment:
I like the idea of having a capped provider rewards but I think this line could cause reward instability. Are you sure that if the reward cap is activated I won't cause a reward drop bigger than if the reward is almost capped?
Why not setting a capped reward at `usersPool[tokenAddress].mul(25).div(100)` ?

Reply:
Because _rewardPercentage is already capped at 25% is the SLORegistry

So at the end of the day, the reward is `25% / leverage * providersPool` if there is enough liquidity in usersPool, or simply **up to** 25% of the usersPool.

Reply:
Based on our discussions there are two main approaches:
* A/ Apply the 25% cap to the usersPool
* B/ Apply the SLI deviation, capped to 25%, to the usersPool

I think A is more of a purist approach, than something that can be market-validated. 

From a user perspective, it's better to lose less, and have a longstanding coverage (through the availability of deposited premiums). I am strongly assuming they will prefer approach B, and our data also shows this.

I'm vouching for keeping the user-centric approach implemented here. At least until v2.0

Reply:
Good, let's resolve this. Specific IT test cases should be added in DTK to address these edge cases.